src/client.rs

```diff
@@ -282,12 +325,27 @@ impl Response {
     }
 }
 
+/// Read the body of the Response, decoding it if necessary.
+impl Read for Decoder {
+    fn read(&mut self, buf: &mut [u8]) -> io::Result<usize> {
+        match self {
+            &mut Decoder::PlainText(ref mut hyper_response) => {
+                hyper_response.read(buf)
+            },
+            &mut Decoder::Gzip(ref mut hyper_response) => {
+                GzDecoder::new(BufReader::new(hyper_response))?.read(buf)
```

Comment:
This `GzDecoder` should be constructed in `from_hyper_response`, and saved. The error you are seeing is that part of the stream is being decoded, and then the decoder is being thrown away. The `BufReader` shouldn't be necessary, because the `hyper::client::Response` already have a `BufReader` inside it.